File: FinalProject/app/src/main/java/com/example/finalproject/service/spell/ManaReservoir.java
package com.example.finalproject.service.spell;

public class ManaReservoir extends Component {
    private final int volume;

    public ManaReservoir(String name, int volume, boolean available) {
        super(name, available);
        this.volume = volume;
    }

    public ManaReservoir(ManaReservoir manaReservoir){
        super(manaReservoir.getName(), manaReservoir.isAvailable());
        volume=manaReservoir.volume;
    }

    public int getVolume() {
        return volume;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/spell/ManaReservoir.kt
package com.example.finalproject.service.spell

class ManaReservoir(val volume:Double, name: String, available: Boolean): Component(name, available) {
    constructor(manaReservoir: ManaReservoir) : this(manaReservoir.volume, manaReservoir.name, manaReservoir.available)
}